Dockerfile

```diff
@@ -48,15 +51,23 @@ RUN \
 --mount=type=bind,source=/opt/NeMo-Curator/pyproject.toml,target=/opt/NeMo-Curator/pyproject.toml,from=curator-update \
   cd /opt/NeMo-Curator && \
   source activate curator && \
-  pip install ".[all]"
+  if [ "$BUILD_TYPE" = "nightly" ]; then \
+    pip install ".[all_nightly]"; \
+  else \
+    pip install ".[all]"; \
+  fi
```

Comment:
We might need to check if the nightly can use the same caching logic as stable, or if nightly should just always build a fresh non-cached container.

Reply:
+1, wanted to flag this explicitly as we do some curator-update optimization above, and I want to ensure if branching doesn't mess with it.
cc @ko3n1g

Reply:
Do these optimizations come from the base-image?

Reply:
I mean, if they do even if the base-image tag is unchanged b/c they overwrite it we should extend this build-workflow so that we can force pull the base image. In all other cases we don't need to be worried about caching i believe